core/src/mindustry/world/blocks/defense/ForceProjector.java

```diff
@@ -32,6 +32,7 @@
     public float cooldownLiquid = 1.5f;
     public float cooldownBrokenBase = 0.35f;
     public float basePowerDraw = 0.2f;
+    private boolean phaseValid;
```

Comment:
This shouldn't be a variable. It's shared for every single block. You can't rely on this being the correct value. Calculate it every time it's needed instead.

Reply:
Is this good now?

Reply:
On second glance, why not use `realRadius(ForceEntity)` instead or calculating it yourself?

Reply:
`realRadius(entity)` would be equivalent to the size of the shield bubble. Since this is already displayed, I opted to show the "max" size, not accounting for overheating or not enough power.